### DISCUSSION TITLE
Filter choices based on value

### DISCUSSION THREAD

**Discussion by danielbrodin** [johnlindquist/kit#141] [Q&A]
Is it possible to set the keys on a `Choice` that should be used to filter the items when using `arg('write something', [])`?
My use case for this is that I have a script that shows all my projects and open them in code, and a choice looks like
```
{
  name: "lorem",
  value: "~/User/something/something/ipsum",
  description: "~/User/something/something",
};
```
and filtering is currently only done on name, but it would be nice to include value as well so I could type `something` which could be a client folder perhaps and `lorem` would show up.

This could perhaps be done by having a `filterKeys: string[]` value on the `PromptConfig`?

**Comment by johnlindquist** [comment]
Hi @danielbrodin , good question (I moved this from an issue to a discussion).

Conceptually, filter "groups" are a good candidate for "tabs":

```js
let directoryGroup = async (name)=> {
  let projects = [] //use "name" to filter/load projects
  await arg("Select project", projects)
}

onTab("Something", async () => {
  await directoryGroup("something")
})

onTab("Dev", async () => {
  await directoryGroup("dev")
})
```

also you could take part of the `value` and append it to the `name`:

```js
await arg("Select project", async () => {
  let directory = "~/projects"
  let files = await readdir(directory)

  return files.map(name => ({
    name: name + " - " + directory,
    value: directory
  }))
})
```

I'm considering adding an `alias` prop for when you want to type a shorthand, but it's not high priority since you can format the name however you want.

**Reply by danielbrodin** [reply]
This is for opening projects I have, so grouping them would make it less productive since it's rarely needed. And having it part of the name would clutter the hole thing up and make it less readable, more so if a path was added after the name.

I understand that it's not a priority, was more of a question if it existed, otherwise a suggestion to improve the filter. At least having the description searchable would make sense imo since it's data that shows.
